woocommerce/woocommerce-gutenberg-products-block
# Introduce a new Products by Tag(s) block

**NOTE: I'm targeting the abstract block refactor branch so I can take advantage of those changes. This will need rebasing with master before merge once the other PR is in.

This PR introduces a products by tag block (based heavily on the products by categories/attributes blocks that exist already).

To prevent scaling issues, if there are large numbers of tags, rather than list all (like the category widget) it will only list all if there are < 20 tags. If there are more, 20 most popular tags will be shown until you search. The search is debounced and based on the code in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/547

Fixes #503 

### Screenshots

![Settings](https://user-images.githubusercontent.com/90977/57367135-54903400-7180-11e9-8171-c8dcf35357d4.png)

### How to test the changes in this Pull Request:

1. Checkout/build branch
2. Ensure you have tags in your catalog (good to test < 20 and > 20 since the search behaviour will be different)
3. Add the "Products by Tag" block to a post and check the options/output.
